### ISSUE TITLE
Sudden hex dump terminal output

### ISSUE THREAD

**Issue by zorael** [ikod/dlang-requests#101]
When streaming from some URLs (such as `https://www.youtube.com`) I sometimes get sudden terminal output that looks like a hex dump. Specifically when foreaching the range that `receiveAsRange` returns for a given `Response`.

I haven't reduced it to a minimal test case yet, I thought I'd ask first if this is intended behaviour? I've tried setting `globalLogLevel(LogLevel.off)`.

Excerpt from the real code:
```d
    Request req;
    req.useStreaming = true;
    req.keepAlive = false;
    req.bufferSize = 8192;

    auto res = req.get(url);
    auto stream = res.receiveAsRange();

    foreach (const part; stream)
    {
        // hex dump sometimes here but not always
        // ...
    }
```

In the terminal:
```
00000  33 EB 58 A3 D4 51 1C 26  0A 95 2B 4C 47 3A 8E 71  |3.X..Q.&..+LG:.q|
00010  45 E9 4C 9E A6 49 AB 54  93 4F D8 ED FB 46 B2 A4  |E.L..I.T.O...F..|
00020  CA 6C 7C 53 5D 82 82 2E  FD 56 30 FA F0 35 F4 20  |.l|S]....V0..5. |
00030  D4 F0 89 B5 81 1E 6F 39  E1 92 1A D3 E1 4D B5 E8  |......o9.....M..|
00040  15 B5 D8 6F 8C DE BE FD  62 FD 49 8A 3C 14 5A AB  |...o....b.I.<.Z.|
00050  F4 11 B3 5B 6B 75 3B AF  D5 C5 E1 8B 97 D6 C1 CB  |...[ku;.........|
00060  F4 7C A5 66 27 35 2E 36  6C 2B 2B 21 31 2D DB 57  |.|.f'5.6l++!1-.W|
00070  C4 41 B2 1D 59 FD 00 87  54 E3 22 A0 DD 5D 34 3F  |.A..Y...T."..]4?|
00080  A2 F9 91 61 2B E8 AE 64  54 96 C0 91 A8 51 07 70  |...a+..dT....Q.p|
00090  A1 9B 54 45 D2 85 A4 74  4A 69 F7 CB 69 99 9C 5C  |..TE...tJi..i..\|
000A0  C0 A9 A0 A7 03 23 41 63  6A A5 04 E9 E8 CE A5 6D  |.....#Acj......m|
000B0  E4 F4 43 8B 92 28 94 43  10 24 84 51 39 44 F7 F7  |..C..(.C.$.Q9D..|
000C0  A8 52 51 D0 33 54 61 A9  A5 5B E9 24 98 F6 93 50  |.RQ.3Ta..[.$...P|
000D0  30 5E 87 37 8C 40 9E BE  00 10 18 B6 D7 46 77 93  |0^.7.@.......Fw.|
000E0  2A 09 9D 88 3D 4C 60 F1  6A 18 0D DB 68 48 42 3B  |*...=L`.j...hHB;|
[...]
```

**Comment by ikod** [comment]
Hello,
looking into this

**Comment by ikod** [comment]
@zorael 
which version of compiler do you use?

**Comment by ikod** [comment]
Hello @zorael 
Answering your question - no this is not intended behaviour - something wrong here.
I tried to reproduce this with simple code
```
import std.stdio;
import requests;

void main()
{
    Request req;
    string url = "https://www.youtube.com";
    req.useStreaming = true;
    req.keepAlive = false;
    req.bufferSize = 8192;

    auto res = req.get(url);
    auto stream = res.receiveAsRange();

    foreach (const part; stream)
    {
        // hex dump sometimes here but not always
        // ...
        writefln("%d", part.length);
    }
}
```
but without success. So, to move forward I'd like to have more info:
1. OS
1. D compiler and version
1. is your application single or multithreaded
1. It would be nice if you can provide code sample so that I can reproduce problem, but this is not required.

**Comment by zorael** [comment]
Hello, sorry for the delayed reply. I tried reducing it and got a working example, but I am very confused and can't say for sure anymore that it is dlang-requests that is causing it...

Machine is running Manjaro/Arch x86_64, dmd is v2.085.1, ldc 1.15.0, requests 1.0.7. The original program is multithreaded but the example is singlethreaded.

```sh
git clone https://github.com/zorael/tests -b hexdump
cd tests
dub run
```

It uses Adam D. Ruppe's [`dom.d`](https://github.com/adamdruppe/arsd/blob/master/dom.d) to parse what requests streams.

* If I remove the parsing and just print the streamed data to screen, there is no hex dump. (--> suggests `arsd.dom` is the cause)
* If I copy some streamed data output into an enum and feed it into the parsing (bypassing the stream `foreach`), there is no hex dump. (--> suggests `requests` is the cause)
* Hex dumps occur between `foreach` bodies, such as it would if it occured in the stream range's `opApply`(?). (--> suggests `requests` or D itself is the cause)

**Comment by ikod** [comment]
@zorael 

Thanks, with your code I can reproduce this strange behaviour!
Will continue tomorrow morning.

**Comment by ikod** [comment]
Hello @zorael 

Bug in initialisation for streaming fixed in 1.0.8 release.
Check it, please, and thanks for your report!

**Comment by zorael** [comment]
It seems to work now with no hex dumps in 1.0.8. Thank you!
